### DISCUSSION TITLE
Feedback and questions for Chapter 2

### DISCUSSION THREAD

**Discussion by antonvorobets** [fortitudo-tech/pcrm-book#34] [Q&A]
This thread is for general feedback and questions related to Chapter 2.

It is perfectly fine to also create threads for specific topics or methods. If they happen to be particularly Chapter 2 related, I will collect references to them here.

**Comment by vico** [comment]
In `chapter2_3.ipynb` the realized implied volatility is calculated by this formula: `realized_vols[i] = 100 * np.sqrt(252 * np.mean(daily_rets[i:i+21]**2))` . I tried using `std` function to do the same but did not get same result, then I noticed the formula do not subtract mean value from daily_rets. This is quite new concept to me so I wonder if this formula is missing the mean or it is what it is?
Thank you for the book and the codes!

**Comment by vico** [comment]
I have another question. In annualized the volatility, looks like we are trying to calculate for a consecutive of 21 days (~ 1 month), and get a monthly volatility? Should we use sqrt(12) for annualizing that monthly volatility ?

**Comment by antonvorobets** [comment]
Thanks for the questions, @vico. I am happy to see that you are carefully studying the code.

What you see in the code is how the realized variance is computed for a variance swap contract: https://en.wikipedia.org/wiki/Variance_swap

<img width="1775" height="371" alt="image" src="https://github.com/user-attachments/assets/d0b21783-616c-4c26-b5c5-6b7f7a56a701" />

If you look at the formula, you should be able to see that it is the same, given that daily_rets are log-returns :-)

Have you seen the Applied Quantitative Investment Management course that goes through the book: https://antonvorobets.substack.com/t/course ?

**Comment by vico** [comment]
No, I have not. Will definitely take the course. Thank you so much for you answer! I learnt something new today.
